# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmBCClass.h
# Commit: 339e37a04ca1
# Full Hash: 339e37a04ca17a209114f9d346751e6d05f6a37d
# Author: Lars T Hansen <lhansen@mozilla.com>
# Date: 2021-10-18 16:09:29
# Regressor Bug: 1727084
# File Overlap Count: 1
# Description:
#   Bug 1727084 - Memory64 - Preliminaries. r=yury
#   
#   Scaffolding: Add assertions everywhere execution currently depends on
#   a memory being a memory32.  These assertions will disappear by and by.
#   
# ==============================================================================

diff -r 83e52246d0ea -r 339e37a04ca1 js/src/wasm/WasmBCClass.h
--- a/js/src/wasm/WasmBCClass.h	Mon Oct 18 09:58:13 2021 +0000
+++ b/js/src/wasm/WasmBCClass.h	Mon Oct 18 09:58:13 2021 +0000
@@ -320,7 +320,10 @@
 
   inline const FuncType& funcType() const;
   inline const TypeIdDesc& funcTypeId() const;
+  inline bool usesMemory() const;
   inline bool usesSharedMemory() const;
+  inline bool isMem32() const;
+  inline bool isMem64() const;
 
   // The casts are used by some of the ScratchRegister implementations.
   operator MacroAssembler&() const { return masm; }